lib/system/supports-colors.js

```diff
@@ -49,12 +45,16 @@ module.exports = (function () {
     return true;
   }
 
-  if (process.env.TERM === 'dumb') {
+  if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
+    return true;
+  }
+
+  if (process.stdout && !process.stdout.isTTY) {
     return false;
   }
 
-  if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
-    return true;
+  if (process.env.TERM === 'dumb') {
```

Comment:
The `process.stdout` check should not have been moved, see my other comment above. When you move it back, the remaining change is that the two `process.env.TERM` switch place, which does nothing. Which means there is nothing left this patch does right. Sorry to say that.